radix-engine-interface/src/blueprints/access_controller/invocations.rs

```diff
@@ -4,36 +4,28 @@ use crate::blueprints::resource::*;
 use crate::*;
 use sbor::rust::fmt::Debug;
 
+pub const ACCESS_CONTROLLER_BLUEPRINT: &str = "AccessController";
+
 //=================================
 // Access Controller Create Global
 //=================================
 
-#[derive(Debug, Clone, Eq, PartialEq, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
-pub struct AccessControllerCreateGlobalInvocation {
-    pub controlled_asset: BucketId,
+pub const ACCESS_CONTROLLER_CREATE_GLOBAL_IDENT: &str = "create_global";
```

Comment:
Thoughts on using enum and statically derive the string using `strum`? or have an additional enum capturing all these variants.

I'm slightly worried that costing and dispatcher can easily go out of sync.

Reply:
Also, how do you think of traits, which has been fantastic for static invocation so far?

```rust
trait NativeFunctionInput: ScryptoEncode + ScryptoDecode {
     const package_address: PackageAddress;
     const fn_ident: &'str;
     type Output;
} 
```

In `PackageApi`, we have two methods 
```rust
trait PackageApi {
    fn call_function(address, ident, args) -> Vec<u8>;
    
    fn call_function_typed<T: NativeFunctionInput>(input) -> T:Output {
         let output = self.call_function(input.package_address, input.fn_ident, scrypto_encode(&input));
         scrypto_decode(&output)
    } 
}
```

Reply:
For Scrypto blueprint in future, we may automatically implement this trait for the stubs as well.

Reply:
hrm, I've thought about it but decided not to go down enum path as that would go back to treating the set of native methods as a closed set rather than an open set, which seems to be against the new philosophy. I think the syncing issue between these identifiers should be resolved by having additional state per method rather then relying on an enum. Happy to chat about this though.

Reply:
The traits stuff looks cool! Will save that though for a future PR since it doesn't affect the underlying architecture.

Reply:
On the other hand, rather than relying on traits I think we should think about how this looks like with code generation reliant on abis.